pkg/apis/workflow/v1alpha1/generated.proto

```diff
@@ -859,6 +859,10 @@ message RetryStrategy {
 
   // Backoff is a backoff strategy
   optional Backoff backoff = 3;
+
+  // If exists, ScheduleOnDifferentHostNodes prevents of running workflow step on the same host
+  // Step won't rerun on the host where label is equal scheduleOnDifferentHostNodesLabel, e.g. "kubernetes.io/hostname" or "k3s.io/hostname"
+  optional string scheduleOnDifferentHostNodesLabel = 4;
```

Comment:
@jessesuen thoughts on the proto?

Reply:
could we use something like a NodeSelector instead of string?

Reply:
A label like `"k3s.io/hostname"` it's unique across all nodes in the cluster.
With `NodeSelector` we could use any label but this not give us uniqueness, moreover, this will complicate this PR. We have already to use the field `NodeStatus.HostNodeName`.

Reply:
maybe

```yaml
retryStrategy:
    nodeAffinity:
      differentHost: true  
      nodeAffinity: ...
```

Reply:
@jessesuen bump.